sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -159,6 +262,97 @@ string_class ProgramManager::getProgramBuildLog(const RT::PiProgram &Program) {
   return Log;
 }
 
+struct ImageDeleter {
+  void operator()(DeviceImage *I) {
+    delete[] I->BinaryStart;
+    delete I;
+  }
+};
+
+ProgramManager::ProgramManager() {
+  const char *SpvFileC = std::getenv(UseSpvEnv.c_str());
+  // If a SPIRV file is specified with an environment variable,
+  // register the corresponding image
+  if (SpvFileC)
+    SpvFile = SpvFileC;
+  if (!SpvFile.empty()) {
+    // The env var requests that the program is loaded from a SPIRV file on disk
+    std::ifstream File(SpvFile, std::ios::binary);
+
+    if (!File.is_open())
+      throw runtime_error(std::string("Can't open file specified via ") +
+                          UseSpvEnv + ": " + SpvFile);
+    File.seekg(0, std::ios::end);
+    size_t Size = File.tellg();
+    auto *Data = new unsigned char[Size];
+    File.seekg(0);
+    File.read(reinterpret_cast<char *>(Data), Size);
+    File.close();
+
```

Comment:
"Data" leakes if exception happens here.

Reply:
Although ifstream does not throw exceptions by default, so this code looks safe, I much prefer using unique_ptr to avoid manual delete below, applied.

Reply:
Thanks.